File: docs/samplegeneratedcode/java/sampletargetj/IService2ProxyContext.java
package sampletargetj;        
// Factory generated on Tue Jul 02 09:12:42 BST 2024
//   by com.typedpath.unittestof.processor.TestableRoots2Kotlin
//   for com.typedpath.unittestof.annotationprocessor.UnitTestOfAnnotationProcessor    

import com.typedpath.unittestof.testutil.CallCentre;
import com.typedpath.unittestof.testutil.CallCentre.Match;
import com.typedpath.unittestof.testutil.CallCentre.MatchRow;
public class IService2ProxyContext extends  com.typedpath.unittestof.testutil.ServiceProxyContext<IService2> {
public IService2ProxyContext(CallCentre callCentre, String id) {
   super(sampletargetj.IService2.class, callCentre, id);
}
    public void whenSayHello2(Match<String> name ,   String result) {
       getCallCentre().add(new MatchRow(getId(), "sayHello2_String", java.util.Arrays.asList(name), result));
    }
    public void whenSayHello2(Match<String> name, Match<String> lastName ,   String result) {
       getCallCentre().add(new MatchRow(getId(), "sayHello2_String_String", java.util.Arrays.asList(name, lastName), result));
    }
    public java.util.List<String> captureSayHello2(Match<String> name ) {
       return getCallCentre().getArgumentsByProxyIdMethodId(getId(), "sayHello2_String").stream().map (it -> (String) it[0] ).collect(java.util.stream.Collectors.toList()); 
    }         

    public static class sayHello2_String_StringParams {
    public String name;
    public String lastName;
    public sayHello2_String_StringParams (String name, String lastName) {
      this.name=name;
      this.lastName=lastName;    
    }
    } 
    public java.util.List<sayHello2_String_StringParams> captureSayHello2(Match<String> name, Match<String> lastName ) {
       return getCallCentre().getArgumentsByProxyIdMethodId(getId(), "sayHello2_String_String").stream().map (it-> new sayHello2_String_StringParams((String) it[0],(String) it[1])  ).collect(java.util.stream.Collectors.toList()); 
    }    
        

}
    

File: docs/samplegeneratedcode/kotlin/targetsample/IService2ProxyContext.kt
package targetsample        
// Factory generated on Tue Jul 02 09:08:31 BST 2024
//   by com.typedpath.unittestof.processor.TestableRoots2Kotlin
//   for UnitTestOfKspSymbolProcessor    

import com.typedpath.unittestof.testutil.CallCentre
import com.typedpath.unittestof.testutil.CallCentre.Match
import com.typedpath.unittestof.testutil.CallCentre.MatchRow
@Suppress("unused")
class IService2ProxyContext(callCentre: CallCentre, id: String) : com.typedpath.unittestof.testutil.ServiceProxyContext<IService2>(targetsample.IService2::class, callCentre, id) {
    @Suppress("UNUSED_PARAMETER")
    fun whenEquals(other: Match<Any> ,  result: kotlin.Boolean) {
       callCentre.add(MatchRow(id, "equals_Any", listOf(other), result))
    }

    @Suppress("UNUSED_PARAMETER")
    fun whenHashCode(  result: kotlin.Int) {
       callCentre.add(MatchRow(id, "hashCode_", listOf(), result))
    }

    @Suppress("UNUSED_PARAMETER")
    fun whenLog(text: Match<String> ,  result: kotlin.String) {
       callCentre.add(MatchRow(id, "log_String", listOf(text), result))
    }

    @Suppress("UNUSED_PARAMETER")
    fun whenLog(severity: Match<String>, text: Match<String> ,  result: kotlin.String) {
       callCentre.add(MatchRow(id, "log_String_String", listOf(severity, text), result))
    }

    @Suppress("UNUSED_PARAMETER")
    fun whenToString(  result: kotlin.String) {
       callCentre.add(MatchRow(id, "toString_", listOf(), result))
    }

        @Suppress("UNUSED_PARAMETER")
    fun captureEquals(other: Match<Any> ) : List<Any> {
       return callCentre.getArgumentsByProxyIdMethodId(id, "equals_Any").map { it[0]   } 
    }     
        

        @Suppress("UNUSED_PARAMETER")
    fun captureHashCode( ) : List<Any> {
       return callCentre.getArgumentsByProxyIdMethodId(id, "hashCode_").map { listOf<Any>()  } 
    }     
        

        @Suppress("UNUSED_PARAMETER")
    fun captureLog(text: Match<String> ) : List<String> {
       return callCentre.getArgumentsByProxyIdMethodId(id, "log_String").map { it[0]  as String  } 
    }     
        

    data class log_String_StringParams( val severity: String, val text: String )        
    @Suppress("UNUSED_PARAMETER")
    fun captureLog(severity: Match<String>, text: Match<String> ) : List<log_String_StringParams> {
       return callCentre.getArgumentsByProxyIdMethodId(id, "log_String_String").map { log_String_StringParams(it[0]  as String,it[1]  as String)  } 
    }
        

        @Suppress("UNUSED_PARAMETER")
    fun captureToString( ) : List<Any> {
       return callCentre.getArgumentsByProxyIdMethodId(id, "toString_").map { listOf<Any>()  } 
    }     
        

}
    